includes/Service/class-omnisendapiservice.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Omnisend Api service
+ *
+ * @package OmnisendPaidMerbershipsProPlugin
+ */
+
+declare(strict_types=1);
+
+namespace Omnisend\PaidMembershipsProAddon\Service;
+
+use Omnisend\PaidMembershipsProAddon\Actions\OmnisendAddOnAction;
+use Omnisend\PaidMembershipsProAddon\Mapper\ContactMapper;
+use Omnisend\PaidMembershipsProAddon\Validator\ResponseValidator;
+use Omnisend\SDK\V1\Omnisend;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Omnisend API Service.
+ */
+class OmnisendApiService {
+	/**
+	 * Contact mapper.
+	 *
+	 * @var ContactMapper
+	 */
+	private $contact_mapper;
+
+	/**
+	 * Omnisend client
+	 *
+	 * @var Omnisend
+	 */
+	private $client;
+
+	/**
+	 * Response validator
+	 *
+	 * @var ResponseValidator
+	 */
+	private $response_validator;
+
+	/**
+	 * OmnisendApiService class constructor.OmnisendLifterLMSPlugin
+	 */
+	public function __construct() {
+		$this->contact_mapper     = new ContactMapper();
+		$this->response_validator = new ResponseValidator();
+		$this->client             = Omnisend::get_client(
+			OMNISEND_MEMBERSHIPS_ADDON_NAME,
+			OMNISEND_MEMBERSHIPS_ADDON_VERSION
+		);
+	}
+
+	/**
+	 * Creates an Omnisend contact.
+	 *
+	 * @param array $form_data The form data.
+	 *
+	 * @return array Tracker data.
+	 */
+	public function create_omnisend_contact( array $form_data ): array {
+		$contact  = $this->contact_mapper->get_omnisend_contact( $form_data );
+		$response = $this->client->create_contact( $contact );
+
+		if ( ! $this->response_validator->is_valid( $response ) ) {
+			return array();
+		}
+
+		return array(
+			OmnisendAddOnAction::EMAIL        => $form_data['bemail'],
+			OmnisendAddOnAction::PHONE_NUMBER => $form_data['bphone'],
+		);
+	}
+
+	/**
+	 * Creates an Omnisend contact.
+	 *
+	 * @param array $form_data The form data.
+	 *
+	 */
+	public function create_omnisend_profile_contact( array $form_data ): void {
+		$current_user = wp_get_current_user();
+		$user_email   = $current_user->user_email;
+		$response     = $this->client->get_contact_by_email( $user_email );
+		$phone_number = $response->get_contact()->get_phone();
```

Comment:
why are we using phone here instead email?

Reply:
PMP module profile form does not have phone number field. So we get phone from already existing Omnisend contact by doing so we can allow both consent checkbox for email and for phone if phone exist in contact.